openshift_resources/resource.py

```diff
@@ -0,0 +1,50 @@
+from abc import ABCMeta
+from typing import Optional, Any
+
+from kubernetes.dynamic import DynamicClient
+from ocp_resources.resource import _get_client, _get_api_version
+from pydantic import BaseModel, Field
+from resources.io.k8s.apimachinery.pkg.apis.meta.v1 import ObjectMeta
+
+
+class BaseResource(BaseModel, metaclass=ABCMeta):
```

Comment:
since name + namespace (or just the name for cluster resource) a unique ID for a resource in k8,
I suggest adding a name & namespace